Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=6c0735687d37e25a65866823881bcbf39a6a023f
Commit:     6c0735687d37e25a65866823881bcbf39a6a023f
Parent:     889394b1eb95f040525d06dd4f2f1222b94023e2
Author:     Sebastian Andrzej Siewior <bigeasy@linutronix.de>
AuthorDate: Sun Nov 30 16:50:04 2008 +0100
Committer:  Greg Kroah-Hartman <gregkh@suse.de>
CommitDate: Wed Jan 7 10:00:04 2009 -0800

    USB: isp1760: use a specific PLX bridge instead of any bdridge
    
    this driver can't handle (of course) any brdige class devices. So we
    now are just active on one specific bridge which should be only the
    isp1761 chip behind a PLX bridge.
    
    Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
    Tested-by: Karl Bongers <kblists08@gmail.com>
    Cc: stable <stable@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

 drivers/usb/host/isp1760-if.c |   16 ++++++++++------
 1 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/drivers/usb/host/isp1760-if.c b/drivers/usb/host/isp1760-if.c
index b87ca7c..a53114c 100644
--- a/drivers/usb/host/isp1760-if.c
+++ b/drivers/usb/host/isp1760-if.c
@@ -268,12 +268,16 @@ static void isp1761_pci_shutdown(struct pci_dev *dev)
 	printk(KERN_ERR "ips1761_pci_shutdown\n");
 }
 
-static const struct pci_device_id isp1760_plx [] = { {
-	/* handle any USB 2.0 EHCI controller */
-	PCI_DEVICE_CLASS(((PCI_CLASS_BRIDGE_OTHER << 8) | (0x06 << 16)), ~0),
-		.driver_data = 0,
-},
-{ /* end: all zeroes */ }
+static const struct pci_device_id isp1760_plx [] = {
+	{
+		.class          = PCI_CLASS_BRIDGE_OTHER << 8,
+		.class_mask     = ~0,
+		.vendor		= PCI_VENDOR_ID_PLX,
+		.device		= 0x5406,
+		.subvendor	= PCI_VENDOR_ID_PLX,
+		.subdevice	= 0x9054,
+	},
+	{ }
 };
 MODULE_DEVICE_TABLE(pci, isp1760_plx);
 
